File: backend/components/metiers/threed/api/samples/samples/sample_edge_cases.py
# sample_edge_cases.py – Exemple ultra avancé de gestion des edge cases (API Threed)
from ..controllers import get_threed
import logging

def sample_edge_cases_ultra():
    # Edge case : accès à un ID inexistant
    try:
        entity = get_threed(-999)
        assert entity is None
        logging.info('Edge case: accès à un ID inexistant OK')
    except Exception as e:
        logging.warning(f"Erreur edge case: {e}")
    print('Sample edge cases ultra avancé exécuté avec succès.')

File: backend/components/metiers/threed/api/samples/samples/sample_edge_cases.js
/* global console */
// sample_edge_cases.js – Exemple ultra avancé de gestion des edge cases (API Threed)
const { getById } = require('../controllers/threed_controller');

function sampleEdgeCasesUltra() {
  // Edge case : accès à un ID inexistant
  try {
    const entity = getById(-999);
    if (entity === null || entity === undefined) {
      console.info('Edge case: accès à un ID inexistant OK');
    } else {
      throw new Error('L’ID inexistant retourne une entité !');
    }
  } catch (e) {
    console.warn('Erreur edge case:', e);
  }
  console.log('Sample edge cases ultra avancé exécuté avec succès.');
}

module.exports = sampleEdgeCasesUltra;
